AppDeployToolkit/AppDeployToolkitExtensions.ps1

```diff
@@ -618,34 +626,61 @@ function Add-NxtXmlNode {
 		$Attributes,
 		[Parameter(Mandatory = $false)]
 		[string]
-		$InnerText
+		$InnerText,
+		[Parameter(Mandatory = $false)]
+		[ValidateSet('Ascii', 'BigEndianUnicode', 'Default', 'Unicode', 'UTF8', 'UTF8withBom')]
+		[string]
+		$Encoding,
+		[Parameter(Mandatory = $false)]
+		[ValidateSet('Ascii', 'BigEndianUnicode', 'Default', 'Unicode', 'UTF8', 'UTF8withBom')]
+		[string]
+		$DefaultEncoding
 	)
 	Begin {
 		## Get the name of this function and write header
-		[string]${CmdletName} = $PSCmdlet.MyInvocation.MyCommand.Name
+		[string]${cmdletName} = $PSCmdlet.MyInvocation.MyCommand.Name
 	}
 	Process {
 		try {
 			if ($false -eq (Test-Path -Path $FilePath)) {
 				Write-Log -Message "File $FilePath does not exist" -Severity 3
 				throw "File $FilePath does not exist"
 			}
-			[xml]$xml = [xml]::new()
-			$xml.Load($FilePath)
+			[hashtable]$importNxtXmlFileParams = @{
+				'Path' = $FilePath
+			}
+			if ($false -eq [string]::IsNullOrEmpty($Encoding)) {
+				$importNxtXmlFileParams['Encoding'] = $Encoding
+			}
+			if ($false -eq [string]::IsNullOrEmpty($DefaultEncoding)) {
+				$importNxtXmlFileParams['DefaultEncoding'] = $DefaultEncoding
+			}
+			[System.Xml.XmlDocument]$xml = Import-NxtXmlFile @importNxtXmlFileParams
 			[string]$parentNodePath = $NodePath.Substring(0, $NodePath.LastIndexOf("/"))
 			if ($true -eq ([string]::IsNullOrEmpty($parentNodePath))) {
 				throw "The provided node root path $NodePath does not exist"
 			}
 			[string]$lastNodeChild = $NodePath.Substring($NodePath.LastIndexOf("/") + 1)
 			# Test for Parent Node
 			if ($false -eq (Test-NxtXmlNodeExists -FilePath $FilePath -NodePath $parentNodePath)) {
-				Add-NxtXmlNode -FilePath $FilePath -NodePath $parentNodePath
-				[xml]$xml = [xml]::new()
-				$xml.Load($FilePath)
+				Write-Log -Message "Parent node $parentNodePath does not exist. Creating it." -Source ${cmdletName}
+				[hashtable]$addNxtXmlNodeParams = @{
```

Comment:
This hashtable does not change throughout the function. Keep using $importNxtXmlFileParams. Maybe give it a more general name?

Reply:
Sth like "$encodingParams" and use the others directly with the function call